power-the-polls-form/src/components/ptp-info-jurisdiction/AdditionalInfoForm.tsx

```diff
@@ -0,0 +1,232 @@
+import { Fragment, h } from "@stencil/core";
+import { FunctionalComponent } from "@stencil/router/dist/types/stencil.core";
+
+import { PtpFormData } from "../../util";
+import { submitToActionKit } from "../power-the-polls-form/ActionKit";
+
+type OptionProps = {
+   type: "radio" | "checkbox" | "select",
+   choices: string[],
+   inputName: string,
+};
+const Options: FunctionalComponent<OptionProps> = ( { choices, inputName, type }, children ) =>
+   type === "select"
+      ? (
+         <label>
+            {children}
+            <select name={inputName} required>
+               {choices.map( x => <option value={x}>{x}</option> )}
+            </select>
+         </label>
+      ) : (
+         <div>
+            <label>{children}</label>
+            <div class={{ "option-grid": choices.length > 3 }}>
+               {choices.map( x =>
+                  <label key={x} class="checkbox">
+                     <input
+                        type={type}
+                        name={inputName}
+                        value={x}
+                        required={type === "radio" ? true : false}
+                     />{" "}{x}
+                  </label> )}
+            </div>
+         </div>
+      );
+
+/**
+ * Additional info be collected after users sign up
+ */
+const AdditionalInfoForm: FunctionalComponent<{ data: PtpFormData, onSubmit: () => void }> = ( { data, onSubmit } ) => {
```

Comment:
Are any of the fields in this form required ?

Reply:
They all are